src/physics/thermal_conduction.cpp

```diff
@@ -51,7 +51,7 @@ void ThermalConduction::setTemperature(mfem::Coefficient& temp)
 void ThermalConduction::setTemperatureBCs(const std::set<int>&               temp_bdr,
                                           std::shared_ptr<mfem::Coefficient> temp_bdr_coef)
 {
-  bcs_.addEssential(temp_bdr, temp_bdr_coef, temperature_);
+  bcs_.addEssential(temp_bdr, temp_bdr_coef, temperature_, 0);
```

Comment:
I'm not sure we want to require this. It makes sense that a scalar coefficient on a scalar finite element state would not need a component index.

Reply:
I think that's reasonable, it looks like the condition that should be changed then is here:
https://github.com/LLNL/serac/blob/9fc81ff51ce6a4b85a82e88e593f3bdc88eba771/src/physics/utilities/boundary_condition.cpp#L51-L63

I think the correct modification is to always assume scalar coef when the FEState's dimension is zero - does that sound right to you?

Reply:
Sounds good to me!